solidity/test/bridge/Bridge.test.ts

```diff
@@ -968,1024 +1023,1171 @@ describe("Bridge", () => {
-        const { walletPubKeyHash } = data.deposits[0].reveal
+      })
+    })
 
-        before(async () => {
-          await createSnapshot()
+    context("when the wallet state is MovingFunds", () => {
+      // The execution of `submitSweepProof` is the same for wallets in
+      // `MovingFunds` state as for the ones in `Live` state. Therefore the
+      // testing of `MovingFunds` state is limited to just one simple test case
+      // (sweeping single P2SH deposit).
+      let tx: ContractTransaction
+      const data: SweepTestData = SingleP2SHDeposit
+      const { fundingTx, reveal } = data.deposits[0]
 
-          // Simulate the wallet is an Live one and is known in
-          // the system.
-          await bridge.setWallet(walletPubKeyHash, {
-            ecdsaWalletID: ethers.constants.HashZero,
-            mainUtxoHash: ethers.constants.HashZero,
-            pendingRedemptionsValue: 0,
-            createdAt: await lastBlockTime(),
-            moveFundsRequestedAt: 0,
-            state: walletState.Live,
-          })
-        })
+      before(async () => {
+        await createSnapshot()
 
-        after(async () => {
-          await restoreSnapshot()
+        // Initially set the state to Live, so that the deposit can be revealed
+        await bridge.setWallet(reveal.walletPubKeyHash, {
+          ecdsaWalletID: ethers.constants.HashZero,
+          mainUtxoHash: ethers.constants.HashZero,
+          pendingRedemptionsValue: 0,
+          createdAt: await lastBlockTime(),
+          moveFundsRequestedAt: 0,
+          state: walletState.Live,
         })
 
-        it("should revert", async () => {
-          // Corrupt the bitcoin headers length in the sweep proof. The proper
-          // value is length divisible by 80 so any length violating this
-          // rule will cause failure. In this case, we just remove the last
-          // byte from proper headers chain.
-          const properHeaders = data.sweepProof.bitcoinHeaders.toString()
-          data.sweepProof.bitcoinHeaders = properHeaders.substring(
-            0,
-            properHeaders.length - 2
-          )
+        await relay.setCurrentEpochDifficulty(data.chainDifficulty)
+        await relay.setPrevEpochDifficulty(data.chainDifficulty)
 
-          await expect(runSweepScenario(data)).to.be.revertedWith(
-            "Invalid length of the headers chain"
-          )
+        await bridge.revealDeposit(fundingTx, reveal)
+
+        // Simulate the wallet's state has changed to MovingFunds
+        const wallet = await bridge.getWallet(reveal.walletPubKeyHash)
+        await bridge.setWallet(reveal.walletPubKeyHash, {
+          ecdsaWalletID: wallet.ecdsaWalletID,
+          mainUtxoHash: wallet.mainUtxoHash,
+          pendingRedemptionsValue: wallet.pendingRedemptionsValue,
+          createdAt: wallet.createdAt,
+          moveFundsRequestedAt: wallet.moveFundsRequestedAt,
+          state: walletState.MovingFunds,
         })
```

Comment:
That can be simplified to:
```
        await bridge.setWallet(reveal.walletPubKeyHash, {
          ...wallet,
          state: walletState.MovingFunds,
        })
```

Reply:
Done in cbbe1bd2097ee552daf4e4016a7e35fe7620bfe3.